### PR TITLE
Add max mods per peptide to gptmd

### PR DESCRIPTION
**Pull request by zrolfs**

### CODE REVIEW

**Comment by zrolfs** [comment]
#1049 
There's some hacky stuff going on where some notches are combined and others aren't. I left a note in the xaml about it:
"The maximum number of modifications allowed for a single notch. The current implementation does not combine notches EXCEPT for those with masses annotated in "...\\MetaMorpheus\\Data\\combos.txt". In MetaMorpheus near the top left, click "Data", select "Open folder with mods/data files", and open the "Data" folder. Implementing and increasing this parameter may result in high search times and high FDRs."

File: GUI/GPTMDTaskWindow.xaml [hunk 1]
```diff
@@ -49,14 +49,14 @@
                         <StackPanel>
                             <GroupBox Header="Deisotoping">
                                 <StackPanel>
-                                    <CheckBox x:Name="useProvidedPrecursor" Content="Use Provided Precursor" IsChecked="True" ToolTipService.ShowDuration="999999" ToolTipService.InitialShowDelay="500">
+                                    <CheckBox x:Name="UseProvidedPrecursor" Content="Use Provided Precursor" IsChecked="True" ToolTipService.ShowDuration="999999" ToolTipService.InitialShowDelay="500">
```
Comment (acesnik) [inline GUI/GPTMDTaskWindow.xaml]: Yum, I love code style fixes.

File: TaskLayer/GPTMDTask/GPTMDTask.cs [hunk 2]
```diff
@@ -199,8 +199,8 @@ protected override MyTaskResults RunSpecific(string OutputFolder, List<DbForTask
                     var line = r.ReadLine().Split(' ');
                     var mass1 = double.Parse(line[0], CultureInfo.InvariantCulture);
                     var mass2 = double.Parse(line[1], CultureInfo.InvariantCulture);
-                    if (modificationsThatCanBeCombined.Where(b => b.ValidModification == true).Any(b => Math.Abs((double)b.MonoisotopicMass - mass1) < tolForComboLoading) &&
-                        modificationsThatCanBeCombined.Where(b => b.ValidModification == true).Any(b => Math.Abs((double)b.MonoisotopicMass - mass2) < tolForComboLoading))
+                    if (modificationsThatCanBeCombined.Any(b => b.ValidModification == true && Math.Abs((double)b.MonoisotopicMass - mass1) < tolForComboLoading) &&
```
Comment (acesnik) [inline TaskLayer/GPTMDTask/GPTMDTask.cs]: nice

File: GUI/GPTMDTaskWindow.xaml [hunk 3]
```diff
@@ -418,10 +430,27 @@
                                 <ColumnDefinition Width="1*"/>
                             </Grid.ColumnDefinitions>
                             <Grid.RowDefinitions>
+                                <RowDefinition Height="25"/>
                                 <RowDefinition Height="25"/>
                                 <RowDefinition Height="1*"/>
                             </Grid.RowDefinitions>
                             <StackPanel Grid.Column="0" Grid.Row="0" Orientation="Horizontal">
+                                <Label x:Name="MaxModsPerNotchLabel" Content="Max Mods Per Notch" />
```
Comment (acesnik) [inline GUI/GPTMDTaskWindow.xaml]: 👍

File: GUI/GPTMDTaskWindow.xaml [hunk 4]
```diff
@@ -418,10 +430,27 @@
...
+                                <TextBox x:Name="MaxModsPerNotchTextBox" PreviewTextInput="CheckIfNumber" Width="45" ToolTipService.ShowDuration="999999" ToolTipService.InitialShowDelay="500" IsEnabled="False">
```
Comment (rmillikin) [inline GUI/GPTMDTaskWindow.xaml]: it seems kind of weird to have a textbox that's not enabled; can you put the extra info about combos in the tooltip for "Max Mods Per Peptide Before GPTMD"?

**Pull request merged**